driver/src/main/java/oracle/nosql/driver/http/Client.java

```diff
@@ -717,8 +717,10 @@ public Result execute(Request kvRequest) {
                 throw new NoSQLException("Unexpected exception: " +
                         rae.getMessage(), rae);
             } catch (InvalidAuthorizationException iae) {
-                /* allow a single retry on clock skew errors */
-                if (iae.getMessage().contains("clock skew") == false ||
+                /* allow a single retry on clock skew / auth errors */
+                String msg = iae.getMessage();
+                if ((msg.contains("clock skew") == false &&
+                    msg.contains("request signature is invalid") == false) ||
```

Comment:
This would not do the right thing if IAM starts returning expired token exceptions. See the code I posted in slack